api/src/main/java/org/pmiops/workbench/db/dao/UserService.java

```diff
@@ -73,6 +71,30 @@ private void updateDataAccessLevel(User user) {
     }
   }
 
+  public User createUser(String givenName, String familyName, String email, String contactEmail) {
+    User user = new User();
+    user.setDataAccessLevel(DataAccessLevel.UNREGISTERED);
+    user.setEmail(email);
+    user.setContactEmail(contactEmail);
+    user.setFamilyName(familyName);
+    user.setGivenName(givenName);
+    try {
+      userDao.save(user);
+    } catch (DataIntegrityViolationException e) {
+      user = userDao.findUserByEmail(email);
+      if (user == null) {
+        throw e;
+      }
+      // If a user already existed, update it in place.
```

Comment:
I like this logic. I think this type of logic scattered through our code is what is helping keep our dev environment robust.